internal/controller/siteconfig_controller.go

```diff
@@ -187,12 +198,41 @@ func (r *SiteConfigReconciler) handleFinalizer(ctx context.Context, siteConfig *
 	return ctrl.Result{}, false, nil
 }
 
+func (r *SiteConfigReconciler) isSiteConfigValidated(siteConfig *sc_v1alpha1.SiteConfig) bool {
+	condition := meta.FindStatusCondition(siteConfig.Status.Conditions, string(conditions.SiteConfigValidated))
+	if condition != nil && condition.Status == metav1.ConditionTrue {
+		return true
+	}
+	return false
+}
+
+func (r *SiteConfigReconciler) handleValidate(ctx context.Context, siteConfig *sc_v1alpha1.SiteConfig) error {
+	r.Log.Info("Starting validation", "SiteConfig", siteConfig.Name)
+	if err := ValidateSiteConfig(ctx, r.Client, siteConfig); err != nil {
+		r.Log.Error(err, "SiteConfig validation failed due to error", "SiteConfig", siteConfig.Name)
+		conditions.SetStatusCondition(&siteConfig.Status.Conditions,
+			conditions.SiteConfigValidated,
```

Comment:
To the same point as above (about validating every time) I think this should be `SiteConfigValid` rather than `Validated`.

Reply:
In light of the fact that we will keep validating the SiteConfig on every reconcile request (for the SiteConfig in question) until the cluster is provisioned, I will change the condition to `ValidSiteConfig`

Reply:
However, it may be better to refactor the `SiteConfigValidated` to be a boolean field rather than a condition:
```golang
type SiteConfigStatus struct {
	ValidSiteConfig bool `json:"validSiteConfig,omitempty"`
	Conditions []metav1.Condition `json:"conditions,omitempty"`
	ClusterDeploymentRef *corev1.LocalObjectReference `json:"clusterDeploymentRef"`
	DeploymentConditions []hivev1.ClusterDeploymentCondition `json:"deploymentConditions,omitempty"`
	ManifestsRendered []ManifestReference `json:"manifestsRendered,omitempty"`
}

```
@carbonin wdyt?

Reply:
I vaguely recall somewhere in an API style guide for OpenShift the recommendation to not use booleans in the API (focus was on configuration but I think the same rationale applies here). They mention that booleans are inherently restrictive and there is no ability to extend them as use cases grow/evolve. Just a thought...

Reply:
Thanks @imiller0 - since all the conditions are in the past tense, it feels awkward having one of them in the present tense (i.e. `SiteConfigValid` or `ValidSiteConfig`). Furthermore, the condition reasons are predominantly in the past tense as well:
```golang
const (
	Completed  ConditionReason = "Completed"
	Failed     ConditionReason = "Failed"
	TimedOut   ConditionReason = "TimedOut"
	InProgress ConditionReason = "InProgress"
	Unknown    ConditionReason = "Unknown"
)
```
imo we can still keep the `SiteConfigValidated` as is since it is a condition that speaks to the act of "validating" a SiteConfig. At the time which the condition is updated on the SiteConfig - the action would have been completed.

Reply:
I think if we look at the current set of conditions they are all "phases of operation" for the controller. In that light SiteConfigValidated makes the most sense (to me). Nick makes a good point that a common pattern for conditions is also to identify key conditions of interest, such as "the SiteConfig is valid" and "deployment completed". In the short run perhaps we:
1. Keep SiteConfigValidated as-is because it aligns with the other phase conditions
2. Consider an additional set of conditions as an improvement which capture these key states (SiteConfigOK, DeploymentCompleted, etc

Reply:
Thanks Ian! Going with Option 1 for now as we can always revisit the conditions at a later stage in a subsequent PR.